packages/pubsub/test/subscription.js

```diff
@@ -372,922 +225,1279 @@ describe('Subscription', function() {
+          throw new Error('Should not be called.');
+        };
+
+        subscription.flushQueues_();
       });
 
-      it('should invoke callback with error from ack', function(done) {
-        var error = new Error('Error.');
-        subscription.ack = function(id, callback) {
+      it('should emit any connection acquiring errors', function(done) {
+        var error = new Error('err');
+
+        subscription.connectionPool.acquire = function(callback) {
           callback(error);
         };
-        subscription.pull({}, function(err) {
-          assert.equal(err, error);
+
+        subscription.on('error', function(err) {
+          assert.strictEqual(err, error);
           done();
         });
-      });
 
-      it('should execute callback', function(done) {
-        subscription.pull({}, done);
+        subscription.flushQueues_();
       });
 
-      it('should return pull response as apiResponse', function(done) {
-        var resp = {
-          receivedMessages: [{
-            ackId: 1,
-            message: {
-              messageId: 'abc',
-              data: new Buffer('message').toString('base64')
-            }
-          }]
+      it('should write the acks to the connection', function(done) {
+        fakeConnection.write = function(reqOpts) {
+          assert.deepEqual(reqOpts.ackIds, ['abc', 'def']);
+          done();
         };
 
-        subscription.ack = function(id, callback) {
-          callback(null, { success: true });
-        };
+        subscription.flushQueues_();
+      });
 
-        subscription.parent.request = function(protoOpts, reqOpts, callback) {
-          callback(null, resp);
+      it('should write the nacks to the connection', function(done) {
+        fakeConnection.write = function(reqOpts) {
+          assert.deepEqual(reqOpts.modifyDeadlineAckIds, ['ghi', 'jkl']);
+          assert.deepEqual(reqOpts.modifyDeadlineSeconds, [0, 0]);
+          done();
         };
 
-        subscription.pull({}, function(err, msgs, apiResponse) {
-          assert.deepEqual(resp, apiResponse);
-          done();
-        });
+        subscription.flushQueues_();
       });
-    });
-  });
 
-  describe('seek', function() {
-    it('should throw if a name or date is not provided', function() {
-      assert.throws(function() {
-        subscription.seek();
-      }, /Either a snapshot name or Date is needed to seek to\./);
+      it('should clear the inventory after writing', function() {
+        subscription.flushQueues_();
+
+        assert.strictEqual(subscription.inventory_.ack.length, 0);
+        assert.strictEqual(subscription.inventory_.nack.length, 0);
+      });
     });
 
-    it('should make the correct api request', function(done) {
-      var FAKE_SNAPSHOT_NAME = 'a';
-      var FAKE_FULL_SNAPSHOT_NAME = 'a/b/c/d';
+    describe('without connection pool', function() {
+      it('should do nothing if theres nothing to ack/nack', function() {
```

Comment:
We have this test twice. I would move it outside of the describe so it's not associated with or without a connection pool.